preppal-fe/src/pages/recipe-calendar.tsx

```diff
@@ -0,0 +1,236 @@
+            setRecipeID("");
+            setRecipeTitle("");
+            setShoppingList("");
+          }
+        }
+        else {
+          // default to empty
+          setRecipeID("");
+          setRecipeTitle("");
+          setShoppingList("");
+        }
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  async function getOwnRecipes() {
+    const token = sessionStorage.getItem("token");
+    let fetchedRecipes: any[] = [];
+    try {
+      if (token) {
+        const req = {
+          method: "GET",
+          headers: {
+            'x-auth-token': token
+          }
+        };
+        fetchedRecipes = await fetch("http://localhost:9001/api/users/ownRecipes/", req).then((res) => res.json());
+      }
+      // return a this users' recipes
+      return fetchedRecipes;
+    }
+    catch (err) {
+      console.log(err);
+      return fetchedRecipes;
+    }
+  }
+
+  // When a calendar date is selected, update currDay and recipeOfTheDay if exists
+  async function updateDayAndRecipe(value: Value) {
+    onChange(value);
+    if (calendarObject && calendarObject.calendarDates && calendarObject.calendarDates.length > 0 && value) {
+      const dateIs = value?.toLocaleString().split(",")[0];
+      const index = calendarObject.calendarDates.findIndex((calDate: any) => calDate.dateIs === dateIs);
+      if (index > -1) {
+        // If the day exists in the calendar, set global variables with calendar variables
+        setRecipeID(calendarObject.calendarDates[index].recipeOfTheDayID);
+        setRecipeTitle(calendarObject.calendarDates[index].recipeOfTheDayTitle);
+        setShoppingList(calendarObject.calendarDates[index].recipeOfTheDayIngredients);
+      }
+      else {
+        // set to default
+        setRecipeID("");
+        setRecipeTitle("");
+        setShoppingList("");
+      }
+    }
+    else {
+      // set to default
+      setRecipeID("");
+      setRecipeTitle("");
+      setShoppingList("");
+    }
+  }
+
+  // runs when a recipe is selected for a day in the calendar
+  const updateRecipeAndShoppingList = (event: any) => {
+    if (event && event?.target && ownRecipes) {
+      const thisID = (event.target as HTMLInputElement).value
+      // using thisID for a recipe, get the recipe index and set global variables
+      const index = ownRecipes.findIndex((recipe: any) => recipe._id === thisID);
+      if (index > -1) {
+        // if valid ID in ownRecipes, set global variables
+        setRecipeID(ownRecipes[index]._id);
+        setRecipeTitle(ownRecipes[index].title);
+        const shoppingList = ownRecipes[index].ingredients.join(", ");
+        setShoppingList(shoppingList);
+      }
+      else {
+        // set to default
+        setRecipeID("");
+        setRecipeTitle("");
+        setShoppingList("");
+      }
+    }
+  }
+
+  // when update calendar day clicked, submit current recipe and date to api to update backend
+  async function submitDay() {
```

Comment:
would it be possible to have some sort of user notification to let them know if the transaction was successful?
something like `Recipe added to {date}` or `Recipe could not be added to {date}`

display the message however you want. popup, alert, or text next to/below the button

Do this it only if it is an easy addition. otherwise, make this an enhancement for next sprint.